File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] temp = new int[nums.Length];
        temp[0] = 1;
        int pre = nums[0];
        int pos = 1;

        for (int i = 1; i < nums.Length; i++) {
            temp[i] = pre;
            pre *= nums[i];
        }

        for (int i = nums.Length-1; i >= 0; i--) {
            temp[i] *= pos;
            pos *= nums[i];
        }
        return temp;
    }
}

//[1,-1,-1,0,0], 
//[ , , ,0,0]

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        totalProd = 1
        zeros = 0
        temp = 1
        res = []
        for n in nums:
            if n is not 0:
                totalProd = totalProd * n
            else:
                zeros = zeros + 1

        # print(res)
        if zeros > 1:
            for n in nums:
                res.append(0)
            return res
        elif zeros > 0:
            temp = 0
            
        for n in nums:
            # print(n)
            if (n is 0):
                res.append(totalProd)
            else:
                prod = totalProd // n * temp
                res.append(prod)
        return res